postgres/datadog_checks/postgres/query_count_cache.py

```diff
@@ -0,0 +1,36 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from cachetools import TTLCache
+
+# 8 hours
+CACHE_TTL = 60 * 60 * 8
+
+
+class QueryCountCache:
+    """Maintains a cache of the last-known number of calls per queryid"""
+
+    def __init__(self):
+        self.cache = TTLCache(
+            maxsize=10000,
+            ttl=CACHE_TTL,
+        )
+        self.update_counts = {}
+
+    def set_calls(self, queryid, calls):
+        """Updates the cache of calls per query id.
+
+        Returns whether or not the number of calls changed based on
+        the newly updated value. The first seen update for a queryid
+        does not count as a change in values since that would result
+        in an inflated value."""
+        calls_changed = False
+        if queryid in self.cache:
+            diff = calls - self.cache[queryid]
+            calls_changed = diff > 0
```

Comment:
There's a small edge case not covered by `diff > 0`. If a query is evicted from `pg_stat_statements` and then executed again at a later time, the count will be reset and diff will be negative. I think we should treat this as `calls_changed = True`.

Reply:
Yes you're right, I meant to add a test case for this. Adding now.